components/elm/src/biogeophys/SurfaceAlbedoMod.F90

```diff
@@ -1589,5 +1619,559 @@ subroutine TwoStream(bounds, &
+    real(r8) :: f_rdir, f_rdir_temp                       ! adjustment factor for reflected-direct flux
+    real(r8) :: f_dif                                     ! adjustment factor for diffuse flux
+    real(r8) :: f_rdif, f_rdif_temp                       ! adjustment factor for reflected-diffuse flux
+    real(r8) :: fd_prime                                  ! temp adjustment factor for direct flux
+    real(r8) :: fi_prime                                  ! temp adjustment factor for diffuse flux
+    real(r8) :: albd_adjust                               ! adjusted albedo for direct flux
+    real(r8) :: albi_adjust                               ! adjusted albedo for diffuse flux
+    real(r8) :: ftemp1, ftemp2, dzen1, dzen2
+    real(r8) :: local_timeofday                           ! local time of day (second)
+    real(r8) :: coeff_dir(3,0:7)                          ! regression coefficients for f_dir
+    real(r8) :: coeff_rdir(3,0:7)                         ! regression coefficients for f_rdir
+    real(r8) :: coeff_dif(4,0:7)                          ! regression coefficients for f_dif
+    real(r8) :: coeff_rdif(3,0:7)                         ! regression coefficients for f_rdif
+
+    data coeff_dir(:,1) /2.045E+1, 6.792E-1, -2.103E+1/
+    data coeff_dir(:,2) /1.993E+0, 9.284E-1, -2.911E+0/
+    data coeff_dir(:,3) /5.900E-2, 9.863E-1, -1.045E+0/
+    data coeff_dir(:,4) /5.270E-3, 9.942E-1, -9.995E-1/
+    data coeff_dir(:,5) /2.977E-3, 9.959E-1, -9.990E-1/
+    data coeff_dir(:,6) /2.977E-3, 9.959E-1, -9.990E-1/
+    data coeff_dir(:,7) /8.347E-3,        0, -8.393E-3/
+
+    data coeff_rdir(:,1) / 2.351E-1, 1.590E-1, -2.332E-1/
+    data coeff_rdir(:,2) / 1.368E-1, 1.642E-1, -1.358E-1/
+    data coeff_rdir(:,3) / 1.254E-1, 1.653E-1, -1.247E-1/
+    data coeff_rdir(:,4) / 1.274E-1, 1.635E-1, -1.267E-1/
+    data coeff_rdir(:,5) / 1.314E-1, 1.623E-1, -1.307E-1/
+    data coeff_rdir(:,6) / 1.359E-1, 1.620E-1, -1.352E-1/
+    data coeff_rdir(:,7) /-4.463E-6, 1.556E-1,  1.287E-3/
+
+    data coeff_dif(:,1) /3.146E-7, 4.385E+0, 6.723E-3, -4.382E+0/
+    data coeff_dif(:,2) /6.001E-7, 4.068E+0, 2.456E-2, -4.085E+0/
+    data coeff_dif(:,3) /7.436E-7, 3.911E+0, 5.606E-2, -3.960E+0/
+    data coeff_dif(:,4) /7.806E-7, 3.763E+0, 1.049E-1, -3.863E+0/
+    data coeff_dif(:,5) /7.581E-7, 3.559E+0, 1.734E-1, -3.727E+0/
+    data coeff_dif(:,6) /7.015E-7, 3.298E+0, 2.543E-1, -3.547E+0/
+    data coeff_dif(:,7) /6.359E-7, 2.984E+0,        0, -2.984E+0/
+
+    data coeff_rdif(:,1) / 1.493E-1, 1.621E-1, -1.483E-1/
+    data coeff_rdif(:,2) / 1.462E-1, 1.654E-1, -1.454E-1/
+    data coeff_rdif(:,3) / 1.454E-1, 1.673E-1, -1.446E-1/
+    data coeff_rdif(:,4) / 1.465E-1, 1.683E-1, -1.457E-1/
+    data coeff_rdif(:,5) / 1.443E-1, 1.682E-1, -1.435E-1/
+    data coeff_rdif(:,6) / 1.446E-1, 1.686E-1, -1.439E-1/
+    data coeff_rdif(:,7) /-3.427E-6, 1.576E-1,  1.199E-3/
```

Comment:
need to add `_r8` to all numbers i.e. `2.045E+1_r8`

Reply:
I see. Will revise them